src/azure-cli/azure/cli/command_modules/batch/_client_factory.py

```diff
@@ -89,12 +87,7 @@ def batch_data_service_factory(cli_ctx, kwargs):
     if not account_key:
         from azure.cli.core._profile import Profile
         profile = Profile(cli_ctx=cli_ctx)
-        # in order to use AAD auth in cloud shell mode, we will use mgmt AAD token
-        # instead of Batch AAD token to auth
-        if in_cloud_console():
-            resource = cli_ctx.cloud.endpoints.active_directory_resource_id
-        else:
-            resource = cli_ctx.cloud.endpoints.batch_resource_id
```

Comment:
May I ask why has this logic changed?

Reply:
The logic was added in this PR: https://github.com/Azure/azure-cli/pull/6582, I asked the PR owner, and he seems to forget the details. Maybe it's been broken for a long time and no one spotted it. Using the management endpoint token will result in customer isn't able to authenticate with our data plane, which generally only accepts Batch endpoint resource tokens. I already tested it in the cloud shell environment and verified this fix will resolve the customer's error.